src/test/java/teammates/test/cases/action/AdminEmailComposeSaveActionTest.java

```diff
@@ -0,0 +1,231 @@
+
+        data = (AdminEmailComposePageData) pageResult.data;
+        assertNull(data.emailToEdit);
+
+        ______TS("case: save existing email failure: invalid subject");
+        email = adminEmailsLogic.getAdminEmailDrafts().get(0);
+        content = "valid content";
+        subject = " ";
+        receiver = "test@example.tmt";
+        action = getAction(Const.ParamsNames.ADMIN_EMAIL_CONTENT, content,
+                Const.ParamsNames.ADMIN_EMAIL_SUBJECT, subject,
+                Const.ParamsNames.ADMIN_EMAIL_ADDRESS_RECEIVERS, receiver,
+                Const.ParamsNames.ADMIN_EMAIL_ID, email.emailId);
+        pageResult = getShowPageResult(action);
+        assertEquals(
+                Const.ViewURIs.ADMIN_EMAIL + "?error=true&user=admin.user", pageResult.getDestinationWithParams());
+
+        expectedLogSegment = Const.ACTION_RESULT_FAILURE;
+        AssertHelper.assertContains(expectedLogSegment, action.getLogMessage());
+
+        expectedStatus = "The provided email subject is not acceptable to TEAMMATES as it contains only whitespace"
+                + " or contains extra spaces at the beginning or at the end of the text.";
+        AssertHelper.assertContains(expectedStatus, pageResult.getStatusMessage());
+
+        data = (AdminEmailComposePageData) pageResult.data;
+        assertEquals(subject, data.emailToEdit.subject);
+        assertEquals(email.emailId, data.emailToEdit.emailId);
+
+        ______TS("case: save existing email failure: invalid content");
+        email = adminEmailsLogic.getAdminEmailDrafts().get(0);
+        content = "";
+        subject = "valid subject";
+        receiver = "test@example.tmt";
+        action = getAction(Const.ParamsNames.ADMIN_EMAIL_CONTENT, content,
+                Const.ParamsNames.ADMIN_EMAIL_SUBJECT, subject,
+                Const.ParamsNames.ADMIN_EMAIL_ADDRESS_RECEIVERS, receiver,
+                Const.ParamsNames.ADMIN_EMAIL_ID, email.emailId);
+        pageResult = getShowPageResult(action);
+        assertEquals(
+                Const.ViewURIs.ADMIN_EMAIL + "?error=true&user=admin.user", pageResult.getDestinationWithParams());
+
+        expectedLogSegment = Const.ACTION_RESULT_FAILURE;
+        AssertHelper.assertContains(expectedLogSegment, action.getLogMessage());
+
+        expectedStatus = "email content should not be empty.";
+        AssertHelper.assertContains(expectedStatus, pageResult.getStatusMessage());
+
+        data = (AdminEmailComposePageData) pageResult.data;
+        assertEquals(subject, data.emailToEdit.subject);
+        assertEquals(email.emailId, data.emailToEdit.emailId);
+
+        ______TS("case: save non-existing email success");
+        String emailId = "nonExisitingId";
+        content = "valid content";
+        subject = "valid subject <b>To check sanitization</b>";
+        receiver = "test@example.tmt";
+        action = getAction(Const.ParamsNames.ADMIN_EMAIL_CONTENT, content,
```

Comment:
Consider moving `Const.ParamsNames.ADMIN_EMAIL_CONTENT, content,` to the line below (in line with the other parameters).